scripts/scripts-1.js

```diff
@@ -0,0 +1,171 @@
+function getMinNumberFromParams(numb_1, numb_2, numb_3, numb_4) {
+
+  const minimalNumber = Math.min(numb_1, numb_2, numb_3, numb_4);
+
+  if (numb_1 === undefined || numb_2 === undefined || numb_3 === undefined || numb_4 === undefined) {
+
+    return 'Undefined value!';
+
+  } else if (numb_1 === '' || numb_2 === '' || numb_3 === '' || numb_4 === '') {
+
+    return 'Your value is empty';
+
+  } else if (isNaN(minimalNumber)) {
+
+    return 'Your value is not a number!';
+
+  }
+
+  else
+
+    return minimalNumber;
+
+};
+
+function greatestСommonDivisor(numb_1, numb_2) {
+
+  const remainder = numb_1 % numb_2;
+
+  if (numb_2 === undefined || numb_1 === undefined) {
+
+    return 'Undefined value!';
+
+  } else if (isNaN(numb_2) || isNaN(numb_1)) {
+
+    return 'Your value is not a number!';
+
+  } else if (numb_2 !== 0) {
+
+    return greatestСommonDivisor(numb_2, remainder);
+  }
+
+  return numb_1;
+};
+
+function perfectNumber(numb) {
```

Comment:
Same thing with validation and calculations.
Plus, it should validate strings
<img width="566" alt="image" src="https://user-images.githubusercontent.com/84010501/191556840-99503a25-7dba-4d4b-99c1-7d8f95691f9f.png">